app/api/views/alias.py

```diff
@@ -318,6 +319,25 @@ def update_alias(alias_id):
         alias.pinned = data.get("pinned")
         changed = True
 
+    if "enable_SMTP" in data:
+        enable_SMTP = data.get("enable_SMTP")
+        if enable_SMTP:
+            if "smtp_password" in data:
+                smtp_password = data.get("smtp_password")
+                if (
+                    smtp_password and len(smtp_password) != 21
+                ):  # 21 is default password length set for nanoid.
+                    return jsonify(error="Invalid SMTP Password Length"), 400
+                smtp_cred = SMTPCredentials.create(alias_id=alias.id, password=smtp_password)
+                if not smtp_cred:
+                    return jsonify(error="Error Saving SMTP Password"), 400
+            else:
+                return jsonify(error="SMTP password not supplied"), 400
+        else:
+            SMTPCredentials.delete_by_alias_id(alias_id=alias.id)
+        alias.enable_SMTP = enable_SMTP
+        changed = True
```

Comment:
Instead of generating the password in the client it'd be better to generate it server-side and show it once to the client on successful activation:

 - No need to add extra dependencies in python and in js (ie. nanoid)
 - No need to have js code in the front
 - No strange 21 character validation
 - Other clients have no need to add nanoid dependencies.
 
Just generate a random secret, store it in the db and in the next page display show the credentials warning the user that the credentials won't be able to be displayed any more times.

Reply:
@nguyenkims Is this really needed at this stage ?

Reply:
@sahilph  generally I think it's better to generate secret on server as we can't control client environment. Adding this change should be quite simple right as the server generates the password and send this back to the client to display?